helm-chart-sources/logstream-leader/templates/_pod.tpl

```diff
@@ -22,6 +22,9 @@ containers:
     {{- range $key, $value := .Values.securityContext }}
     {{- if or (eq $key "runAsUser") (eq $key "runAsGroup") (eq $key "fsGroup")}}
       {{ $key }}: {{ $value | int }}
+    {{- else if kindIs "map" $value }}
```

Comment:
Please add UT in

```
./helm-chart-sources/logstream-leader/tests/securitycontext_test.yaml
```

Just for inspiration:
```
  - it: Should handle the conversion of capabilities from map to array format
    set:
      securityContext:
        runAsUser: 1000
        capabilities:
          drop:
            - ALL
    asserts:
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            capabilities:
              drop:
                - ALL
            runAsUser: 1000
```

You can consider to add another fixtures 
```
./helm-charts/helm-chart-sources/logstream-workergroup/tests/fixtures
```
To cover your change as well